firefox-ios/Client/Frontend/Browser/BrowserViewController/Actions/ActionProviderBuilder.swift

```diff
@@ -0,0 +1,196 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import WebKit
+import Photos
+import Shared
+
+class ActionProviderBuilder {
```

Comment:
This is a builder, but it doesn't use a fluent API (i.e., methods don't return `self`). Since almost all options are applied conditionally, I believe a fluent API wouldn't be appropriate here.